lib/runners/workspace/git.rb

```diff
@@ -0,0 +1,49 @@
+module Runners
+  class Workspace::Git < Workspace
+    private
+
+    def prepare_base_source(dest)
+      base = git_source.base
+      if base
+        provision(base, dest)
+      else
+        raise ArgumentError, "base must not be nil"
+      end
+    end
+
+    def prepare_head_source(dest)
+      provision(git_source.head, dest)
+    end
+
+    def provision(commit_hash, dest)
+      shell = Shell.new(current_dir: git_directory, trace_writer: trace_writer, env_hash: {})
+      shell.capture3!("git checkout #{commit_hash}")
```

Comment:
Is this also forgotten?

```suggestion
      shell.capture3!("git", "checkout", commit_hash)
```

Reply:
Thank you :+1: